README.md

```diff
@@ -46,6 +46,24 @@ watcher.on('delete', function (filepath, root) { console.log('file deleted', fil
 watcher.close();
 ```
 
+In addition to change events, the watcher may emit a `fresh_instance` event that
+signal the state got lost and we will not receive notifications for some of the
+changes; notably, file deletions. The recommended approach when this happens is
+to recrawl the filesystem according with your application logic, then continue
```

Comment:
I would suggest invalidating caches here; the fresh instance result from watchman will include the full crawl results anyway.  The key thing about this signal is that you should invalidate everything that you may have cached that was not included in the file list that had `is_fresh_instance: true`.

Since we're emitting the fresh instance event prior to passing on the list of files you may want to advise folks to invalidate everything they have cached and let them know that the event will be immediately followed by the full list of files.